tc/sgx/trusted_worker_manager/enclave/wpe/CMakeLists.txt

```diff
@@ -66,14 +66,23 @@ LINK_ENCLAVE_COMMON_WORKLOAD_LIBRARIES()
 # ADDITIONAL LIBRARIES
 # XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
-# Add echo workload
-TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -lecho_wo -Wl,--no-whole-archive)
-# Add heart disease eval workload
-TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -lheart_disease_eval -Wl,--no-whole-archive)
-# Add example inside out workload
-TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -linside_out_eval -Wl,--no-whole-archive)
-# Add simple wallet workload
-TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -lsimple_wallet -Wl,--no-whole-archive)
+FOREACH(workload ${WORKLOAD_LIST})
```

Comment:
Can we create a CMakeWorkloads.txt and create 2 lists.
Workload ids List 1 -> All the workload ids supported by Avalon
Workload library List2 -> Corresponding libraries of workload ids
Note: We don't have map data structure in cmake so as an alternate approach we can maintain two parallel list

1) When user passes workload list as build argument for WPE or Singleton, first compare each workload in user list with List1 and get the index of workload. 
2) Using this index get the corresponding library name from List2
3) Call TARGET_LINK_LIBRARIES with above workload library name. 
4) Repeat above step 1,2,3 for each workload id passed as build argument

Reply:
Done